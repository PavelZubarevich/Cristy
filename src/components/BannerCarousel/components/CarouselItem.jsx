import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { IoArrowForward } from 'react-icons/io5';

function CarouselItem({
  title, description, linkText, image, route,
}) {
  return (
    <div className="carouselItem">
      <div className="carouselItem__content">
        <h2 className="carouselItem__title">{title}</h2>
        <p className="carouselItem__description">{description}</p>
        <Link to={route} className="carouselItem__link">
          {linkText}
          <IoArrowForward />
        </Link>
      </div>
      <div className="carouselItem__image">
        <img src={image} alt={title} />
      </div>
    </div>
  );
}

export default CarouselItem;

CarouselItem.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  linkText: PropTypes.string,
  image: PropTypes.string,
  route: PropTypes.string,
};

CarouselItem.defaultProps = {
  title: '',
  description: '',
  linkText: '',
  image: '',
  route: '',
};
