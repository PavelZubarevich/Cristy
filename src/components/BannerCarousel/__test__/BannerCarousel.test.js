import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import CarouselItem from '../components/CarouselItem';

describe('CarouselItem', () => {
  it('should render carousel item', () => {
    render(
      <CarouselItem
        title="title"
        description="description"
        linkText="link text"
        image=""
        route="/test"
      />,
      { wrapper: BrowserRouter },
    );
    expect(document.body).toMatchSnapshot();
  });
});
