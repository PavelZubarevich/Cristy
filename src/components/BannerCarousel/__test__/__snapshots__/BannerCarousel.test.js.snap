// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselItem should render carousel item 1`] = `
<body>
  <div>
    <div
      class="carouselItem"
    >
      <div
        class="carouselItem__content"
      >
        <h2
          class="carouselItem__title"
        >
          title
        </h2>
        <p
          class="carouselItem__description"
        >
          description
        </p>
        <a
          class="carouselItem__link"
          href="/test"
        >
          link text
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M268 112l144 144-144 144m124-144H100"
              fill="none"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="48"
            />
          </svg>
        </a>
      </div>
      <div
        class="carouselItem__image"
      >
        <img
          alt="title"
          src=""
        />
      </div>
    </div>
  </div>
</body>
`;
